bumble/transport/pyusb.py

```diff
@@ -245,6 +258,13 @@ def device_path(device):
         raise ValueError('device not found')
     logger.debug(f'USB Device: {device}')
 
+    # Power Cycle the device
+    try:
+        device = _power_cycle(device)  # type: ignore
```

Comment:
Power-cycling should be optional (controlled with a transport property for example) rather than the default. For USB dongles where firmware is loaded dynamically, it's disruptive to have to re-load the firmware each time you open the device.
Ideally, USB devices are well-behaved, and an HCI RESET should put them back in a clean state.

Reply:
I agree that it should be optional, and adding a specific spec to the transport would make more sense, any specific character you would like to add to the transport specification ?

The primary purpose of this PR is to address the issue that we couldn't return the device to a clean state using an HCI_RESET when it was crashing. We had to resort to manually unplugging and reconnecting the device.

However, oddly enough, the firmware did not reload into the dongles after performing a power cycle through this interface.

![pyusb_power_cycle](https://github.com/google/bumble/assets/30467496/da79be5a-d6ee-4410-8603-2e760b863180)

Reply:
A simple way to explicitly ask that the transport try a power cycle at open could be: prefix the name with `!`.
So: `pyusb:0` --> no power-cycle, `pyusb:!0` --> power-cycle.

Reply:
I've added this logic in the last commit. Let me know if it is okay with you